### PR TITLE
Updates dec 2023

### PR DESCRIPTION
**Pull request by evildmp**

### CODE REVIEW

File: colophon.rst [hunk 1]
```diff
@@ -0,0 +1,56 @@
+Colophon
+===================
```
Comment (ezio-melotti) [inline colophon.rst:2]:
```suggestion
Colophon
========
```

File: colophon.rst [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+Diátaxis is the work of `Daniele Procida <https://vurt.eu>`_.
+
+It has been developed over a number of years, and continues to be elaborated and explored.
+
+
+.. _contact:
+
+Contact me
+---------------------
```
Comment (ezio-melotti) [inline colophon.rst:12]:
```suggestion
Contact me
----------
```

File: colophon.rst [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+The original context for the Diátaxis approach was limited to software product documentation. 
+
+In 2021 I was awarded a Fellowship of the `Software Sustainability Institute <http://software.ac.uk/blog/sorry-state-usable-software-open-science-results-open-science-retreat>`_, to explore its application in scientific research contexts. More recently I've explored its application in internal corporate documentation, organisational management and education, and also its application at scale. This work is on-going.
+
+Other people have corresponded with me to share their experience of applying Diátaxis to note-taking systems and even as part of a systematic approach to household management.
+
+
+Citation and contribution  
+---------------------------------
```
Comment (ezio-melotti) [inline colophon.rst:37]:
```suggestion
Citation and contribution  
-------------------------
```

File: complex-hierarchies.rst [hunk 4]
```diff
@@ -5,73 +5,77 @@ Diátaxis in complex hierarchies
...
+        Tutorial              <- landing page
+            Part 1
+            Part 2
+            Part 3
+        How-to guides         <- landing page
+            Install
+            Deploy
+            Scale
+        Reference             <- landing page
+            Commandline tool
```
Comment (ezio-melotti) [inline complex-hierarchies.rst:27]:
```suggestion
            Command-line tool
```

File: complex-hierarchies.rst [hunk 5]
```diff
@@ -5,73 +5,77 @@ Diátaxis in complex hierarchies
...
                 Linux container
-            deploy
-            scale
-        reference             <- landing page
-            commandline tool
-            available endpoints
+            Deploy
+            Scale
+        Reference             <- landing page
+            Commandline tool
```
Comment (ezio-melotti) [inline complex-hierarchies.rst:64]:
```suggestion
            Command-line tool
```

File: complex-hierarchies.rst [hunk 6]
```diff
@@ -5,73 +5,77 @@ Diátaxis in complex hierarchies
...
+            Security overview
+            Performance
+
+
+Contents pages 
+--------------
 
-Once again, each level of the hierarchy in a section has an overview landing
-page, for orientation.
+Contents pages - typically a home page and any landing pages - provide an oversight of the material they encompass.
```
Comment (ezio-melotti) [inline complex-hierarchies.rst:76]:
```suggestion
Contents pages - typically a home page and any landing pages - provide an overview of the material they encompass.
```
Did you mean "overview" here?

File: explanation.rst [hunk 7]
```diff
@@ -1,25 +1,26 @@
 .. _explanation:
 
-About explanation
+Explanation
 =================
```
Comment (ezio-melotti) [inline explanation.rst:4]:
```suggestion
Explanation
===========
```

File: explanation.rst [hunk 8]
```diff
@@ -128,31 +103,53 @@ guides should reflect this; you should be able to place an implicit (or even exp
 title. For example: *About user authentication*, or *About database connection policies*.
 
 
-Discuss alternatives and opinions
+Admit opinion and perspective
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (ezio-melotti) [inline explanation.rst:107]:
```suggestion
Admit opinion and perspective
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: explanation.rst [hunk 9]
```diff
@@ -128,31 +103,53 @@ guides should reflect this; you should be able to place an implicit (or even exp
...
+Opinion might seem like a funny thing to introduce into documentation. The fact is that all human activity and knowledge is invested within opinion, with beliefs and thoughts. The reality of any human creation is rich with opinion, and that needs to be part of any understanding of it. 
 
+Similarly, any understanding comes from a perspective, a particular stand-point - which means that other perspectives and stand-points exist. **Explanation can and must consider alternatives**, counter-examples or multiple different approaches to the same question.
 
-Don't instruct, or provide technical reference
+In explanation, you're not giving instruction or describing facts - you're opening up the topic for consideration. It helps to think of explanation as discussion: discussions can even consider and weigh up contrary *opinions*.
+
+
+Keep explanation closely bounded
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (ezio-melotti) [inline explanation.rst:117]:
```suggestion
Keep explanation closely bounded
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: foundations.rst [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+
+Diátaxis is successful because it *works* - both users and creators have a better experience of documentation as a result. It makes sense and it feels right.
+
+However, that's not enough to be confident in Diátaxis as a theory of documentation. As a theory, it needs to show *why* it works. It needs to show that there is actually some reason why there are exactly four kinds of documentation, not three or five. It needs to demonstrate rigorous thinking and analysis, and that it stands on a sound theoretical foundation.
+
+Otherwise, it will be just another useful heuristic approach, and the strongest claim we can make for it is that "it seems to work quite well".
+
+
+Two dimensions of craft
+--------------------------------
```
Comment (ezio-melotti) [inline foundations.rst:12]:
```suggestion
Two dimensions of craft
-----------------------
```

File: foundations.rst [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+
+
+Action/cognition
+~~~~~~~~~~~~~~~~
+
+A skill or craft or practice contains both **action** (practical knowledge, knowing *how*, what we do) and **cognition** (theoretical knowledge, knowing *that*, what we think). The two are completely bound up with each other, but they are counterparts, wholly distinct from each from each, two different aspects of the same thing.
+
+
+Acquisition/application
+~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (ezio-melotti) [inline foundations.rst:28]:
```suggestion
Acquisition/application
~~~~~~~~~~~~~~~~~~~~~~~
```

File: how-to-guides.rst [hunk 12]
```diff
@@ -1,15 +1,15 @@
 .. _how-to:
 
-About how-to guides
+How-to guides
 ===================
```
Comment (ezio-melotti) [inline how-to-guides.rst:4]:
```suggestion
How-to guides
=============
```

File: how-to-guides.rst [hunk 13]
```diff
@@ -80,35 +85,54 @@ have to start at the beginning of the whole story and take your reader right to
...
-- unnecessary explanation, for example - distracts both you and the user and dilutes the useful power of the guide.
+Address real-world complexity
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+**A how-to guide needs to be adaptable to real-world use-cases**. One that is useless for any purpose except *exactly* the narrow one you have addressed is rarely valuable. You can't address every possible case, so you must find ways to remain open to the range of possibilities, in such a way that the user can adapt your guidance to their needs.
 
-Don't explain concepts
-~~~~~~~~~~~~~~~~~~~~~~~
+Omit the unnecessary
+~~~~~~~~~~~~~~~~~~~~~
```
Comment (ezio-melotti) [inline how-to-guides.rst:99]:
```suggestion
Omit the unnecessary
~~~~~~~~~~~~~~~~~~~~
```

File: how-to-guides.rst [hunk 14]
```diff
@@ -80,35 +85,54 @@ have to start at the beginning of the whole story and take your reader right to
...
+"Actions" in this context includes physical acts, but also thinking and judgement - solving a problem involves thinking it through. A how-to guide should address how the user thinks as well as what the user does.
 
-Omit the unnecessary
-~~~~~~~~~~~~~~~~~~~~~
 
-In how-to guides, **practical usability is more helpful than completeness.** Whereas a tutorial needs to be a complete,
-end-to-end guide, a how-to guide does not. It should start and end in some reasonable, meaningful place, and require
-the reader to join it up to their own work.
+Describe a logical sequence
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (ezio-melotti) [inline how-to-guides.rst:115]:
```suggestion
Describe a logical sequence
~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: index.rst [hunk 15]
```diff
@@ -9,87 +9,115 @@
...
-In Diátaxis, each of these modes (or types) corresponds to a different user
-need. Each fulfils a different purpose and requires a different approach to
-its creation.
+------
 
-.. image:: /images/diataxis.png
-   :alt: Diátaxis
-   :class: wider
+Contents
+-----------------------------------
```
Comment (ezio-melotti) [inline index.rst:43]:
```suggestion
Contents
--------
```

File: map.rst [hunk 16]
```diff
@@ -1,12 +1,30 @@
-.. _needs:
+.. _map:
 
-The map of needs
+The map 
 =============================
```
Comment (ezio-melotti) [inline map.rst:4]:
```suggestion
The map 
=======
```

File: map.rst [hunk 17]
```diff
@@ -25,73 +43,8 @@ themselves working with - why this particular list of content types
...
-Documentation therefore either *contains theoretical (i.e. propositional)
-knowledge* or *describes practical actions*, and is concerned either with
-serving *our acquisition* or *our application* of knowledge. Hence the map,
-across which the four forms of documentation are laid out.
-
-
-
-Characteristics of documentation
+Expectations and guidance
 ----------------------------------------------------
```
Comment (ezio-melotti) [inline map.rst:48]:
```suggestion
Expectations and guidance
-------------------------
```

File: reference-explanation.rst [hunk 18]
```diff
@@ -1,7 +1,7 @@
 .. _reference-explanation:
 
-Explanation and reference
-==========================
+The difference between reference and explanation
+=================================================
```
Comment (ezio-melotti) [inline reference-explanation.rst:4]:
```suggestion
The difference between reference and explanation
================================================
```

File: reference.rst [hunk 19]
```diff
@@ -1,13 +1,14 @@
 .. _reference:
 
-About reference
+Reference
 ======================
```
Comment (ezio-melotti) [inline reference.rst:4]:
```suggestion
Reference
=========
```

File: reference.rst [hunk 20]
```diff
@@ -50,33 +51,31 @@ it describes, which is a powerful way of ensuring that it remains faithfully acc
 
 ===============
 
-Food and cooking
---------------------
 
-Perhaps you might consult an encyclopaedia to read about an ingredient (for example, about
-liquorice).
+Key principles
+----------------------------------------
```
Comment (ezio-melotti) [inline reference.rst:56]:
```suggestion
Key principles
--------------
```

File: reference.rst [hunk 21]
```diff
@@ -50,33 +51,31 @@ it describes, which is a powerful way of ensuring that it remains faithfully acc
...
 
-..  image:: /images/liquorice.png
-    :alt: Wikipedia's entry for liquorice
+Describe and only describe
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (ezio-melotti) [inline reference.rst:59]:
```suggestion
Describe and only describe
~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: theory.rst [hunk 22]
```diff
@@ -0,0 +1,35 @@
...
+Within the discipline of documentation, discourse tends towards the practical and concrete. The approach is generally heuristic: guidelines, rules of thumb, specific imperatives, principles that we know work. 
+
+As practitioners, we have much to say about what to do, and how to do it and how it works, and relatively little to say about *why* it works. Our sense of the right way to do things is largely based on a combination of intuition and experience. The theoretical aspect of the discipline receives far less attention.
+
+Diátaxis aims to place documentation practice on a more rigorous theoretical footing.
+
+----------------
+
+In this section
+----------------
```
Comment (ezio-melotti) [inline theory.rst:23]:
```suggestion
In this section
---------------
```

File: tutorials-how-to.rst [hunk 23]
```diff
@@ -1,6 +1,6 @@
 .. _tutorials-how-to:
 
-What’s the difference between a tutorial and how-to guide?
+The difference between a tutorial and how-to guide
 ===============================================================
```
Comment (ezio-melotti) [inline tutorials-how-to.rst:4]:
```suggestion
The difference between a tutorial and how-to guide
==================================================
```

File: tutorials.rst [hunk 24]
```diff
@@ -1,45 +1,36 @@
 .. _tutorials:
 
-About tutorials
+Tutorials
 ===============
```
Comment (ezio-melotti) [inline tutorials.rst:4]:
```suggestion
Tutorials
=========
```

File: tutorials.rst [hunk 25]
```diff
@@ -1,45 +1,36 @@
...
+It's important to understand that while a student will learn by doing, what the student *does* is not necessarily what they *learn*. Through doing, they will acquire theoretical knowledge (i.e. facts), understanding, familiarity. They will learn how things relate to each other and interact, and how to interact with them. They will learn the names of things, the use of tools, workflows, concepts, commands. And so on.
 
-For a product, a tutorial turns new learners into users. An inadequate tutorial can prevent a project from
-acquiring new users.
 
 =================
 
 
 The tutorial as a lesson
 -------------------------
```
Comment (ezio-melotti) [inline tutorials.rst:27]:
```suggestion
The tutorial as a lesson
------------------------
```

File: tutorials.rst [hunk 26]
```diff
@@ -49,229 +40,221 @@ Obligations of the teacher
...
 * *meaningful* - the pupil needs to have a sense of achievement
 * *successful* - the pupil needs to be able to complete it
 * *logical* - the path that the pupil takes through it needs to make sense
-* *usefully complete* - the pupil must have an encounter with all of the actions, concepts and tools they need to become
-  familiar with
+* *usefully complete* - the pupil must have an encounter with all of the actions, concepts and tools they need to become familiar with
 
 
 The problem of tutorials
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (ezio-melotti) [inline tutorials.rst:54]:
```suggestion
The problem of tutorials
~~~~~~~~~~~~~~~~~~~~~~~~
```

File: tutorials.rst [hunk 27]
```diff
@@ -49,229 +40,221 @@ Obligations of the teacher
...
---------------------
 
-Perhaps you have had the experience of teaching a child to cook, in which case you'll have encountered most of the main
-demands imposed by a tutorial.
+=================
 
-..  image:: /images/anselmo.jpg
-    :alt:
+Key principles
+---------------------------------
```
Comment (ezio-melotti) [inline tutorials.rst:74]:
```suggestion
Key principles
--------------
```

File: tutorials.rst [hunk 28]
```diff
@@ -49,229 +40,221 @@ Obligations of the teacher
...
-    * abstraction, generalisation
-    * explanation
-    * choices
-    * information
 
-Give your learner things to *do*, through which they can learn. Only your pupil can learn. Sadly, however much you desire
-it, you will not be able to learn for your pupil. You cannot make them learn. All you can do is make it so *they* can
-learn.
+Deliver visible results early and often
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (ezio-melotti) [inline tutorials.rst:105]:
```suggestion
Deliver visible results early and often
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: tutorials.rst [hunk 29]
```diff
@@ -49,229 +40,221 @@ Obligations of the teacher
...
-Get the user started
-~~~~~~~~~~~~~~~~~~~~
 
-Your job is to **get the learner started**, not to turn them into an expert. Don’t ever be embarrassed to start right at
-the beginning: a user can skim rapidly over what’s unnecessary, but if they need something and it’s not there, you risk
-losing them altogether. It’s also perfectly acceptable if what you get the beginner to do is not the way an experienced
-person would, or even if it’s not the ‘correct’ way - a tutorial for beginners is not the same thing as a manual for
-best practice.
+Point out what the learner should notice
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (ezio-melotti) [inline tutorials.rst:113]:
```suggestion
Point out what the learner should notice
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: tutorials.rst [hunk 30]
```diff
@@ -49,229 +40,221 @@ Obligations of the teacher
...
+    Explanation is only pertinent at the moment the *user* wants it. It is not for the documentation author to decide. 
 
+Explanation is one of the hardest temptations for a teacher to resist; even experienced teachers find it difficult to accept that their students' learning does not depend on explanation. This is perfectly natural. Once we have grasped something, we rely on the power of abstraction to frame it to ourselves - and that's how we want to frame it to others. Understanding means grasping general ideas, and abstraction is the logical form of understanding - but these are not what we need in a tutorial, and it's not how successful learning or teaching works.
 
-Offer only minimum, necessary, explanation
+One must see it for oneself, to see the focused attention of a student dissolve into air, when a teacher's well-intentioned explanation breaks the magic spell of learning.
+
+
+... and focus on the concrete
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (ezio-melotti) [inline tutorials.rst:170]:
```suggestion
... and focus on the concrete
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: tutorials.rst [hunk 31]
```diff
@@ -49,229 +40,221 @@ Obligations of the teacher
...
+
+Keep up a narrative of expectations: "You will notice that ..."; "After a few moments, the server responds with ...". Show the user actual example output, or even the exact expected output.
+
+If you know know in advance what the likely signs of going wrong are, consider flagging them: "If the output doesn't show ..., you have probably forgotten to ...".
+
+It's helpful to prepare the user for possibly surprising actions: "The command will probably return several hundred lines of logs in your terminal."
 
 
 Ignore options and alternatives
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (ezio-melotti) [inline tutorials.rst:192]:
```suggestion
Ignore options and alternatives
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: tutorials.rst [hunk 32]
```diff
@@ -49,229 +40,221 @@ Obligations of the teacher
...
-(different options for the command you're using, different ways to use the API, different approaches to the task you're
-describing) - ignore them. **Your guidance needs to remain focused on what's required to reach the conclusion**, and
-everything else can be left for another time.
+Your job is to guide the learner to a successful conclusion. There may be many interesting diversions along the way (different options for the command you're using, different ways to use the API, different approaches to the task you're describing) - ignore them. *Your guidance needs to remain focused on what's required to reach the conclusion*, and everything else can be left for another time.
+
+Doing this helps keep your tutorial shorter and crisper, and saves both you and the reader from having to do extra cognitive work.
+
+
+Aspire to perfect reliability
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (ezio-melotti) [inline tutorials.rst:200]:
```suggestion
Aspire to perfect reliability
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Review by ezio-melotti — COMMENTED** [review]
I went through the changes of this PR, fixed a bunch of overlong headers underlines, and a couple of minor nits.

There's another nit that I haven't fixed: ` - ` is used in several places where a ` --- ` should be used instead (Sphinx will render this as an [em dash](https://en.wikipedia.org/wiki/Dash#Em_dash)).  This is easier to fix with a find and replace on your end, whereas you can apply the headers fixes directly from GitHub with a batch commit of my suggestions.

File: colophon.rst (hunk 1 above)
Comment (evildmp) [reply colophon.rst:2]: I wouldn't normally bother tidying up longer underlines, but since you've gone to the trouble, thanks.

**Pull request merged**